src/strands_tools/code_interpreter/agent_core_code_interpreter.py

```
@@ -85,16 +89,24 @@ def __init__(
         self.region = resolve_region(region)
         self.identifier = identifier or "aws.codeinterpreter.v1"
         self.auto_create = auto_create
+        self.persist_sessions = persist_sessions
+        self.verify_session_before_use = verify_session_before_use
 
         # Generate session name strategy
         if session_name is None:
             self.default_session = f"session-{uuid.uuid4().hex[:12]}"
         else:
-            self.default_session = session_name
+            # Clean session name to meet validation: [0-9a-zA-Z]{1,40}
+            self.default_session = session_name.replace("-", "").replace("_", "")[:40]
```

Can we throw an exception instead? This might make debugging harder

---

>  During Troubleshooting, we hit ValidationException errors until we implemented auto cleaning

Can you explain a bit more about this because AC Code Interpreter should never be seeing this session_name as far as I can tell. They would be seeing identifier `self.identifier = identifier or "aws.codeinterpreter.v1"`

---

I'm still not following why we need to clean anything.

The session_name is to benefit the Agent. The idea being an agent can use the list_local_sessions action to determine which sessions it already created in a context rich(er) way. Meaning the Agent can create a session titled "Finance Tracker Session" then use that later even if the context is truncated - where a uuid would be meaningless to the agent.


So the `client.get_session(session_id=session_name)` isn't necessary what would make sense is 

`client.get_session(session_id=SESSION_DICTIONARY[session_name])

so I'm not following why we need to clean anything, the thing provided to get_session should always be the session_id which wouldn't need cleaning since its directly from code interpreter